integration-cli/docker_cli_swarm_test.go

```diff
@@ -242,3 +278,50 @@ func (s *DockerSwarmSuite) TestSwarmServiceWithGroup(c *check.C) {
 	c.Assert(err, checker.IsNil)
 	c.Assert(strings.TrimSpace(out), checker.Equals, "uid=0(root) gid=0(root) groups=10(wheel),29(audio),50(staff),777")
 }
+
+func (s *DockerSwarmSuite) TestSwarmNodeTaskList(c *check.C) {
+	d := s.AddDaemon(c, true, true)
+
+	name := "node_ps_test"
+	out, err := d.Cmd("service", "create", "--replicas=2", "--name", name, "busybox", "top")
+	c.Assert(err, checker.IsNil)
+	c.Assert(strings.TrimSpace(out), checker.Not(checker.Equals), "")
+
+	waitAndAssert(c, defaultReconciliationTimeout, d.checkActiveContainerCount, checker.Equals, 2)
+
+	out, err = d.Cmd("node", "ps", "self")
+	c.Assert(err, checker.IsNil)
+	c.Assert(out, checker.Contains, name+".1")
+	c.Assert(out, checker.Contains, name+".2")
+
+	// Remove one replica to test to make sure ps only shows the correct tasks
+	// with and w/o the all flag set.
+	out, err = d.Cmd("service", "scale", name+"=1")
+	c.Assert(err, checker.IsNil)
+	c.Assert(strings.TrimSpace(out), checker.Not(checker.Equals), "")
+
+	waitAndAssert(c, defaultReconciliationTimeout, d.checkActiveContainerCount, checker.Equals, 1)
+
+	out, err = d.Cmd("node", "ps", "self")
+	c.Assert(err, checker.IsNil)
+	c.Assert(out, checker.Count, name, 1)
+	c.Assert(out, checker.Contains, "Running")
+
+	out, err = d.Cmd("node", "ps", "self", "-f", "desired-state=shutdown")
+	c.Assert(err, checker.IsNil)
+	c.Assert(out, checker.Count, name, 1)
+	c.Assert(out, checker.Contains, "Shutdown")
+
+	out, err = d.Cmd("node", "ps", "self", "-a")
+	c.Assert(err, checker.IsNil)
+	c.Assert(out, checker.Count, name, 2)
+	c.Assert(out, checker.Contains, "Running")
+	c.Assert(out, checker.Contains, "Shutdown")
+
+	// All should trump our filter
+	out, err = d.Cmd("node", "ps", "self", "-a", "-f", "desired-state=running")
+	c.Assert(err, checker.IsNil)
+	c.Assert(out, checker.Count, name, 2)
+	c.Assert(out, checker.Contains, "Running")
+	c.Assert(out, checker.Contains, "Shutdown")
+}
```

Comment:
This looks like it's at least a couple different cases. I think this should be split up into different tests with a descriptive name of each case.